doc/research/software-conflicts.md

```diff
@@ -0,0 +1,94 @@
+
+Therefore, when a software proposal is calculated, Agama could raise a question if needed in order to make possible to finish the proposal without conflicts. The user should be able to answer the question by selecting a solution for each conflict. Then, Agama would try to calculate the proposal again. If the user decides to cancel the question and finish without solving the problems, then Agama would generate an issue indicating the dependency problems. Note that the installation will not start if there is any issue with level error. Consequently, such a dependency issue would prevent the installation to start.
+
+### Solver problems
+
+The current API in *yast-pkg-bindings* only offers methods to call the solver and to get the number of problems:
+
+~~~
+#PkgSolve -> calls to #resolvePool
+#PkgSolveErrors -> returns size of #problems
+~~~
+
+The API has to be extended to allow getting the list of problems and to indicate the selected solutions for each problem, for example:
+
+~~~
+#PkgSolveProblems -> returns the list of problems
+#PkgSetSolveSolutions -> sets the list of solutions
+~~~
+
+### D-Bus API
+
+Agama needs to provide a new type of question for reporting and selecting solutions for each solver problem.
+
+This could be accomplished by adding a new question interface (e.g., *WithSelection*), which allows selecting an option from a set:
+
+~~~
+"org.opensuse.Agama1.Questions.WithSelection"
+    #SelectionGroups a(usa{sv})
+        id (u)
+        description (s)
+        options (a{sv})
+            id (u)
+            description (s)
+            details (s)
+
+    #SelectOptions(options a(uu))
+        groupId (u)
+        optionId (u)
+~~~
+
+This new interface would offer a property `SelectionGroups`, which returns a list of selection groups. Each selection group has a list of options for selection. Moreover, there is a `SelectOptions` method which receives a list of pairs. A pair contains a group id and an option id. If the question is accepted, then a new proposal is calculated with the options provided by `SelectOptions`.
+
+In case of canceling the question, an issue is generated in software D-Bus object.
+
+### Notes
+
+Agama can be easily fixed to show an issue when selecting a pattern produces some conflicts. But the dependency issue reported by Agama is far from ideal. It only indicates the number of conflicts (e.g., *Found 2 dependency issues.*).
+
+The feature for reporting and fixing solver problems could be iteratively improved in several steps:
+
+1) Start by showing the dependency issue when selecting a pattern.
+2) Extend *yast-pkg-bindings* and improve the issue message, showing the details of the problems.
+3) Raise a question when the solver generates problems. This should be done for each call to the solver, not only for selecting patterns.
```

Comment:
I still think that at least in some cases we should offer option to revert new selection to solve issue. Like you add qt..ok...you add gnome and get conflict with option revert that unselects gnome without too much thinking.

Reply:
Yes, I think it is doable for some cases. We could emit a first question for undo the change. If the question is canceled, then we emit a question for solving issues.

I think it is something to evaluate when implementing it, but it would not require nothing new from D-Bus or yast-pkg-bindings.